go/vt/dbconfigs/dbconfigs.go

```diff
@@ -135,69 +136,121 @@ func registerPerUserFlags(dbc *userConfig, userKey string) {
 
 }
 
+// Connector contains Connection Parameters for mysql connection
+type Connector struct {
+	connParams *mysql.ConnParams
+	dbName     string
+	host       string
+}
+
+// New initializes a ConnParams from mysql connection parameters
+func New(mcp *mysql.ConnParams) Connector {
+	return Connector{
+		connParams: mcp,
+	}
+}
+
+// Connect will invoke the mysql.connect method and return a connection
+func Connect(ctx context.Context, c Connector) (*mysql.Conn, error) {
```

Comment:
One last final nit: This should be a method of Connector: `func (c Connector) Connect(ctx context.Context) (*mysql.Conn,error)`